### PR TITLE
ft-getNoteById# Build out GET REST API Endpoint(CRUD) to get a single note

### PR DESCRIPTION
**Pull request by amaechi-chuks**
#### What does this PR do?

This PR create an endpoint for a user to get a single note by an id.

#### Description of tasks to be completed

    * Add an endpoint for a GET request 

    * add unit tests

#### How should this be manually tested

    * Clone the repo: Run git clone https://github.com/sojida/notepads/ in the folder where you want the repo to be saved.

    * Navigate into the newly created folder and install the dependencies: cd folder_name && npm install
    
    * Ensure that postman is installed in your laptop. If its not install check the README.

    * Start the server using npm run start:dev

    * Make a GET request to http://localhost:6000/api/v1/notes/7139d3af-b8b4-44f6-a49f-9305791700f4 on your postman and hit the enter button

    * You get a Json object listing the details of that particular note

#### Screenshot
UNADJUSTEDNONRAW_thumb_48

### CODE REVIEW

**Comment by Chiazokam** [comment]
Hi Chuks,

You may have to pull from 'origin develop' on your local and resolve the conflicts that ensue.
But be sure to follow the structure already existing and make tests to still pass. 
You can then push again.

File: server/src/middlewares/middleware.js [hunk 1]
```diff
@@ -35,6 +36,23 @@ class middleware {
...
+          error: {
+            status: 404,
+            message: 'Invalid Id',
+          },
+        });
+      }
+      return next();
+    });
+  }
 }
```
Comment (sojida) [inline server/src/middlewares/middleware.js]: You can reuse a similar function in this middleware to carry out this operation.

File: server/test/index.test.js [hunk 2]
```diff
@@ -3,6 +3,9 @@ import chai, { expect } from 'chai';
 import chaiHttp from 'chai-http';
 import app from '../src/index';
 
+const NoteId = '7139d3af-b8b4-44f6-a49f-9305791700f4';
+const wrongNoteId = '7139d3af-b8b4-44f6';
+const applicationUrl = '/api/v1/notes/';
```
Comment (sojida) [inline server/test/index.test.js]: While I understand that this aims at reducing redundancy, please can write this in the test and not here. We can also agree to do this as a team. Thanks

**Review by sojida — CHANGES_REQUESTED** [review]

**Comment by Chiazokam** [comment]
I think you're good to go. Go ahead and merge

**Review by sojida — APPROVED** [review]

**Pull request merged**